contrib/manyclangs-setup.sh

```diff
@@ -0,0 +1,208 @@
+    then
+        git clone https://github.com/elfshaker/elfshaker
+    fi
+}
+
+install_elfshaker_build_dependencies() {
+    if command -v elfshaker > /dev/null
+    then
+        return
+    fi
+
+    if ! command -v cargo &>/dev/null
+    then
+        echo "Rust unavailable, installing"
+        echo
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -- -y
+    fi
+
+    if ! command -v cc &> /dev/null
+    then
+        sudo apt install -y gcc
+    fi
+}
+
+build_elfshaker() {
+    (cd elfshaker && cargo build --release && cp target/release/elfshaker "${ELFSHAKER_BIN}")
+}
+
+install_elfshaker() {
+    if [ "$ELFSHAKER_FROM_SOURCE" ]
+    then
+        install_elfshaker_build_dependencies
+        build_elfshaker
+    else
+        ELFSHAKER_TGZ=elfshaker_${ELFSHAKER_BINARY_VERSION}_${ELFSHAKER_ARCH}-unknown-linux-musl.tar.gz
+        wget --continue \
+            https://github.com/elfshaker/elfshaker/releases/download/${ELFSHAKER_BINARY_VERSION}/${ELFSHAKER_TGZ} \
+            https://github.com/elfshaker/elfshaker/releases/download/${ELFSHAKER_BINARY_VERSION}/${ELFSHAKER_TGZ}.sha256sum
+        sha256sum --check ${ELFSHAKER_TGZ}.sha256sum
+        
+        tar --extract --directory $HOME/.local/bin --file ${ELFSHAKER_TGZ} --strip-components=1 elfshaker/elfshaker
+    fi
+}
+
+fetch_llvm() {
+    if [ ! -d llvm-project ]
+    then
+        git clone --no-checkout https://github.com/llvm/llvm-project llvm-project
+    fi
+    (cd llvm-project && git fetch origin)
+}
+
+install_jobserver() {
+    if ! command jobserver &> /dev/null
+    then
+        if [ ! -d jobclient ]
+        then
+            git clone https://github.com/olsner/jobclient
+        fi
+        (cd jobclient && make && cp jobserver "${ELFSHAKER_BIN_DIR}"/jobserver)
+    fi
+}
+
+install_ninja_jobclient() {
+    if ! command -v ninja-jobclient &> /dev/null
+    then
+        if [ ! -d ninja-jobclient ]
+        then
+            git clone --depth 1 https://github.com/stefanb2/ninja ninja-jobclient
+            NINJA_JOBCLIENT_COMMIT=${NINJA_JOBCLIENT_COMMIT-2b9c81c0ec1226d8795e7725529f13be41eaa385}
+            (cd ninja-jobclient && git fetch origin ${NINJA_JOBCLIENT_COMMIT} && git checkout -B ninja-jobclient FETCH_HEAD) || {
+                echo "See https://github.com/ninja-build/ninja/pull/1140"
+                echo "Commit ${NINJA_JOBCLIENT_COMMIT} failed to checkout."
+                echo "It's possible it's gone, or something else is wrong."
+                echo "Look for a branch which includes jobclient functionality"
+                exit 1
+            }
+        fi
+        (cd ninja-jobclient && python3 ./configure.py && ninja && cp ninja ${ELFSHAKER_BIN_DIR}/ninja-jobclient)
+    fi
+}
+
+install_ccache_from_source() {
+    if ! command -v ccache &> /dev/null
+    then
+        if [ ! -d ccache ]
+        then
+            git clone --depth 1 --single-branch --branch v4.6.1 https://github.com/ccache/ccache
```

Comment:
You mention that we use ccache v4.6.1 because it has some fixes in it. Any idea if those fixes are critical to building the packs? Because if they are, with our current detection strategy of `command -v ccache` we might not detect the case where an older ccache is already available in PATH.

Reply:
Good point, I think I will unconditionally build ccache for now, and pin it. I don't have a specific thing I can easily point to, except for experience. I expect this would be unnecessary at some point the future.